diff --git a/init/profile.sh.in b/init/profile.sh.in
index faa468639..64d21dbcd 100644
--- a/init/profile.sh.in
+++ b/init/profile.sh.in
@@ -3,7 +3,11 @@
 # get current shell name by querying shell variables or looking at parent
 # process name
 if [ -n "${BASH:-}" ]; then
-   shell=${BASH##*/}
+   if [ "${BASH##*/}" = 'sh' ]; then
+      shell='sh'
+   else
+      shell='bash'
+   fi
 elif [ -n "${ZSH_NAME:-}" ]; then
    shell=$ZSH_NAME
 else

